mula/scheduler/schedulers/scheduler.py

```diff
@@ -195,6 +183,21 @@ def push_item_to_queue(self, p_item: models.PrioritizedItem) -> None:
             )
             raise queues.errors.NotAllowedError("Scheduler is disabled")
 
+        # Create task
+        #
+        # NOTE: we set the id of the task the same as the p_item, for easier
+        # lookup.
+        task = models.Task(
+            id=p_item.id,
+            scheduler_id=self.scheduler_id,
+            type=self.queue.item_type.type,
+            p_item=p_item,
+            status=models.TaskStatus.PENDING,
+            created_at=datetime.now(timezone.utc),
+            modified_at=datetime.now(timezone.utc),
+        )
+        self.ctx.datastores.task_store.create_task(task)
+
```

Comment:
Refactored to make the sequence pending > queued > dispatched work correctly within the scheduler